tests/test_isgri_plugin.py

```diff
@@ -64,8 +78,8 @@ def test_isgri_dummy(dispatcher_live_fixture, product_type):
 @pytest.mark.isgri_plugin
 @pytest.mark.isgri_plugin_dummy
 @pytest.mark.dependency(depends=["test_default"])
-@pytest.mark.parametrize("product_type", ['isgri_spectrum', 'isgri_image']) #TODO: jemx too, also lightcurve; and also allowed role passing test
-def test_isgri_dummy_many_pointings(dispatcher_live_fixture, product_type):
+@pytest.mark.parametrize("product_type", ['isgri_spectrum', 'isgri_image', 'isgri_lc']) #TODO: allowed role passing test
+def test_isgri_dummy_data_rights(dispatcher_live_fixture, product_type):
```

Comment:
is this the correct name now? since below is test_isgri_dummy_roles_private_data

Reply:
Since the test was getting too long, I started to break it down to make it more readable and understandable.

Reply:
like, one for data_rights and one for max_pointings? that's good, but now:

isgri:

test_isgri_dummy_data_rights (formely test_isgri_dummy_many_pointings)
test_isgri_dummy_roles_private_data

while for jemx:

test_jemx_dummy_many_pointings 
test_jemx_dummy_roles_private_data

are you sure it's how you wanted it?

By the way, these jemx and isgri tests are almost identical, can be parameterized

Reply:
My idea is:

- test_isgri_dummy_data_rights (formely test_isgri_dummy_many_pointings): tests the without passing the token, and using the `integral_data_right` parameter
- test_isgri_dummy_roles_private_data: test using the token and accessing private data
- test_jemx_dummy_roles_public_data: test using the token and accessing public data

I am now still adjusting it.

Ans true, jemx and isgri dummy are identical so this can be parametrized, but perhaps it might be useful to have two separate test modules also for other tests.

Reply:
but which of these test then many pointings condition?

Reply:
Are you referring to the original `many_pointings` test?

As it is now, this test covers also that

Reply:
I think there is some misunderstanding. Look [there](https://github.com/oda-hub/dispatcher-plugin-integral/pull/16#discussion_r629957787)

Most of all there is difference between request parameter requesting private data and role encoded in the token which allows these parameters to be.

So providing token with some roles will have no effect on the request content, only on whether the request will be allowed.

To that extent, `integral_data_right` parameter should always be assumed to be available in some way, sometimes by assuming the default - "public". As it is done in there:

https://github.com/oda-hub/dispatcher-plugin-integral/blob/master/cdci_osa_plugin/osa_dataserve_dispatcher.py#L670

Reply:
Ok, now it is more clear. Sorry I was misunderstanding a bit. 

In any case, the tests, as organized now, cover the cases of with and without token usage, and public and private requests.

What am I also missing?

Reply:
well, I was just a bit confused about the first test being renamed, since it now test two possibilities for max_pointings for fixed data_rights parameter. This is more informative about max_pointings condition than about integral_data_rights. But then it is also relevant for data rights.

I would reorganize a bit, i.e. maybe all tests can be merged in one short parametrize one.
With integral_data_rights from 1. public, 2. private, 3. not set
And max_pointings being <50 or > 50
Oh, and maybe scw_list being small or long.

But then, I am also fine if it is left as it is for now.

Reply:
I was indeed thinking of reorganizing with the data right parametrized.

Good idea also for the `scw_list`